File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Timestamped.java
package com.example.grts;

import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.GraphQLInterface;
import viaduct.java.api.types.NodeCompositeOutput;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.List;

public interface Timestamped extends GraphQLInterface {

    Type<Timestamped> Reflection = Type.ofClass(Timestamped.class);

    final class Fields implements TypeFields<Timestamped> {
        private Fields() {}

        public static final Field<Timestamped> __typename =
                Field.of("__typename", Reflection);
                public static final Field<Timestamped> createdAt =
                                Field.of("createdAt", Reflection);

                public static final Field<Timestamped> updatedAt =
                                Field.of("updatedAt", Reflection);

    }

    String getCreatedAtOrThrow(String alias);
    String getCreatedAtOrThrow();
    String getCreatedAt(String alias);
    String getCreatedAt();
    String getUpdatedAtOrThrow(String alias);
    String getUpdatedAtOrThrow();
    String getUpdatedAt(String alias);
    String getUpdatedAt();

}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Timestamped.kt
@file:Suppress("warnings")

package viaduct.api.grts

interface Timestamped : viaduct.api.types.Interface {
          fun getCreatedAtOrThrow(alias: String?): kotlin.String
          fun getCreatedAtOrThrow(): kotlin.String
          fun getCreatedAt(alias: String?): kotlin.String?
          fun getCreatedAt(): kotlin.String?

          fun getUpdatedAtOrThrow(alias: String?): kotlin.String?
          fun getUpdatedAtOrThrow(): kotlin.String?
          fun getUpdatedAt(alias: String?): kotlin.String?
          fun getUpdatedAt(): kotlin.String?


    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Timestamped> {
        override final val name = "Timestamped"
        override final val kcls = viaduct.api.grts.Timestamped::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Timestamped> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Timestamped> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Timestamped.Reflection)

            final val createdAt: viaduct.api.reflect.Field<viaduct.api.grts.Timestamped> =
                viaduct.api.internal.FieldImpl("createdAt", viaduct.api.grts.Timestamped.Reflection)

            final val updatedAt: viaduct.api.reflect.Field<viaduct.api.grts.Timestamped> =
                viaduct.api.internal.FieldImpl("updatedAt", viaduct.api.grts.Timestamped.Reflection)

    }

}